discovery/gossiper.go

```diff
@@ -1033,10 +1044,13 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []l
 			return nil
 		}
 
-		// If this is a local ChannelUpdate announcement, send it to
-		// our peer.
-		if !nMsg.isRemote {
-			// Get our peer's public key
+		// If this is a local ChannelUpdate without an AuthProof, it means
+		// it is an update to a channel that is not (yet) supposed to be
+		// announced to the greater network. However, our channel counter
+		// party will need to be given the update, so we'll try sending
+		// the update directly to the remote peer.
+		if !nMsg.isRemote && chanInfo.AuthProof == nil {
```

Comment:
This is only the case if the node that sent us the update is _also_ one of the nodes in the OG `ChannelAnnouncement`.

Reply:
So what you are saying is that the `AuthProof` will always be `nil` if it is coming from the local node? And this would be enough:
```go
if !nMsg.isRemote {
    ...send...
}
?
```

Reply:
Yep! Although I guess the extra check can't hurt...may avoid a nil panic in a weird scenario.

Reply:
Ok, will keep as-is then :)